internal/pkg/metadb/record/record.go

```diff
@@ -83,7 +83,16 @@ func (r *Record) Save() ([]datastore.Property, error) {
 // Load implements the Datastore PropertyLoadSaver interface and converts Datastore
 // properties to corresponding struct fields.
 func (r *Record) Load(ps []datastore.Property) error {
-	externalBlob, ps, err := timestamps.LoadUUID(ps, externalBlobPropertyName)
+        // ps_modified has been added to enable backward compatibility by creating 
+        // a new datastore array that replaces legacy "NumberOfChunks" with "ChunkCount". 
+        var ps_modified []datastore.Property
```

Comment:
ditto

Reply:
Although this fixed the backward compatibility issue within Open Saves, we would still need to change our gRPC client to start passing ChunkCount instead of NumberOfChunks. Since we use a rolling deployment strategy, we need a gRPC client that can handle both fields simultaneously.

Reply:
Thinking more about this, this code change avoids the internal error we were seeing but won't really provide data consistency since we always read "ChunkCount" from datastore.

Reply:
I take this back. I didn't realize the data is already loaded at this point. @zaratsian Are you going to modify the implementation and add a test for this?

Reply:
That sounds good - I'm working on a few updates based on the feedback, plus the tests and will update the PR shortly.